packages/router/utils.ts

```diff
@@ -1583,6 +1613,32 @@ export class ErrorResponseImpl implements ErrorResponse {
   }
 }
 
+/**
+ * Utility class we use to hold unwrapped Responses while preserving status
+ * codes and headers.  This is most useful for dataStrategy implementations
+ * where implementors want to use a custom decoding mechanism and return
+ * pre-decoded data while also preserving response meta information
+ */
+export class DecodedResponse {
```

Comment:
New public API - very similar to `ErrorResponse` but this is intended for user-land usage in `dataStrategy` if they need to pre-decode a response - but this then allows us to carry the statusCode and headers along throughout the rest of the data flows.